src/Type/Php/SprintfFunctionDynamicReturnTypeExtension.php

```diff
@@ -43,7 +43,8 @@ public function getTypeFromFunctionCall(
 		$formatType = $scope->getType($args[0]->value);
 
 		if ($formatType instanceof ConstantStringType) {
-			if (preg_match('/^%[0-9]*\.?[0-9]+[bdeEfFgGhHouxX]$/', $formatType->getValue()) === 1) {
+			// The printf format is %[argnum$][flags][width][.precision]
+			if (preg_match('/^%([0-9]+\$)?[0-9]*\.?[0-9]+[bdeEfFgGhHouxX]$/', $formatType->getValue()) === 1) {
```

Comment:
`%0$s` isn't valid, please modify this part of the regex (and a test for this scenario)

```
%(?:[1-9][0-9]*\$)?
```

Thanks :)

Reply:
fixed. instead of changing the suggested regex, I left it as is and if-cased the `0`, because otherwise a `non-empty-string` would have been by the followup logic, instead of the default-return-type